okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -151,6 +149,26 @@ public void onResponse(Call _call, Response response) {
         }
     }
 
+    private InterruptedIOException handleInterruption(SettableFuture<Response> future) {
+        getDelegate().cancel();
+        // Regardless of the cancel above, the call may have succeeded or is going to succeed, and we need to make
+        // sure the response body is closed correctly in those cases.
+        future.addListener(() -> closeResponseBody(future), MoreExecutors.directExecutor());
```

Comment:
Futures.addCallback can make this simpler, onSuccess will provide the Response instance to avoid a Future.get. The callback can be a private enum singleton.